### PR TITLE
feat: data ops image for jupyterhub

### PR DESCRIPTION
**Pull request by Santhin**
changes:
- new jupyterhub target
- snowsql cli support
- code server snowflake addon
- fix taging with commit sha from PR

### CODE REVIEW

File: jupyterlab-dataops/Dockerfile [hunk 1]
```diff
@@ -34,16 +40,69 @@ RUN pip install -r /opt/requirements/requirements.txt
...
 COPY requirements/requirements-bigquery.txt /opt/requirements/
 RUN pip install -r /opt/requirements/requirements-bigquery.txt
 ADD resources/cloudbeaver/conf/initial-data-sources-bigquery.conf.j2 /opt/cloudbeaver/conf/initial-data-sources.conf.j2
 COPY resources/cloudbeaver/set_metadata.py /opt/cloudbeaver/set_metadata.py
 COPY resources/cloudbeaver/set_metadata_run_jupyter.sh /set_metadata_run_jupyter.sh
 RUN chmod +x /set_metadata_run_jupyter.sh
 CMD ["/set_metadata_run_jupyter.sh"]
+
+# Jupyterhub version
+FROM jupyter/minimal-notebook:python-3.8.13 as jupyterhub
```
Comment (p-pekala) [inline jupyterlab-dataops/Dockerfile:58]: Looks like there is a lot of duplication,if it is not possible to reuse the code maybe it is better to have it in separate Dockerfile?

File: jupyterlab-dataops/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+set -e
+mkdir -p /home/jovyan/.snowsql
```
Comment (p-pekala) [inline jupyterlab-dataops/entrypoint.sh:3]: this is snowflake related so it should be under snowsql I think.

File: jupyterlab-dataops/resources/snowsql/config [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+#rolename = defaultrolename
+#proxy_host = defaultproxyhost
+#proxy_port = defaultproxyport
+
+[connections.example]
+#Can be used in SnowSql as #connect example
+
+accountname = accountname
+username = username
+password = password1234
```
Comment (p-pekala) [inline jupyterlab-dataops/resources/snowsql/config:28]: ???

**Review by p-pekala — CHANGES_REQUESTED** [review]

File: jupyterlab-dataops/resources/snowsql/config (hunk 3 above)
Comment (Santhin) [reply jupyterlab-dataops/resources/snowsql/config:28]: This is the default config option created from snowsql cli, In our case we needed to overwrite existing config due to errors with `log_file`, by changing log destination from `../snowsql_rt.log` to  `~/.snowsql/log`

File: jupyterlab-dataops/entrypoint.sh (hunk 2 above)
Comment (Santhin) [reply jupyterlab-dataops/entrypoint.sh:3]: dd1ffe5a7df855f89af33388392c2d7b2c566398

File: jupyterlab-dataops/resources/snowsql/config (hunk 3 above)
Comment (Santhin) [reply jupyterlab-dataops/resources/snowsql/config:28]: I will delete it from config

**Review by p-pekala — APPROVED** [review]

**Pull request merged**